numpy/_core/src/multiarray/_datetime.h

```diff
@@ -252,9 +253,9 @@ convert_datetime_to_pyobject(npy_datetime dt, PyArray_DatetimeMetaData *meta);
 /*
  * Converts a timedelta into a PyObject *.
  *
- * Not-a-time is returned as the string "NaT".
+ * NaT (Not-a-time) is returned as None.
  * For microseconds or coarser, returns a datetime.timedelta.
- * For units finer than microseconds, returns an integer.
+ * For Y/M (nonlinear units), generic units and units finer than microseconds, returns an integer.
```

Comment:
If I were in your shoes, I'd write something like

```suggestion
 * Returns an integer for Y/M (nonlinear units), ns/ps/fs/as (units shorter than microseconds), and generic units.
```